apps/bridge-dapp/src/containers/UploadTxHistoryModal/UploadTxHistoryModal.tsx

```diff
@@ -11,26 +13,98 @@ import {
   FileUploadList,
   Progress,
   getHumanFileSize,
+  notificationApi,
 } from '@webb-tools/webb-ui-components';
-import { CheckboxCircleLine, KeyIcon } from '@webb-tools/icons';
+import {
+  CheckboxCircleLine,
+  KeyIcon,
+  InformationLine,
+} from '@webb-tools/icons';
+import { parseJson } from '../../utils';
+import { twMerge } from 'tailwind-merge';
+import { txArraySchema } from './types';
 
 const UploadTxHistoryModal: FC<{
   isOpen: boolean;
   setIsOpen: (isOpen: boolean) => void;
 }> = ({ isOpen, setIsOpen }) => {
+  const { addTransactions } = useTxClientStorage();
   const [file, setFile] = useState<File | undefined>();
+  const [transactions, setTransactions] = useState<TransactionType[]>([]);
+  const [isError, setIsError] = useState<boolean>(false);
 
-  const handleUpload = useCallback((files: File[]) => {
+  const handleUploadFile = useCallback((files: File[]) => {
     if (files.length) {
       const file = files[0];
       setFile(file);
     }
   }, []);
 
+  const reset = useCallback(() => {
+    setFile(undefined);
+    setTransactions([]);
+    setIsError(false);
+  }, []);
+
   const closeModal = useCallback(() => {
     setIsOpen(false);
-    setFile(undefined);
-  }, [setIsOpen]);
+    reset();
+  }, [setIsOpen, reset]);
+
+  const handleSaveTxHistory = useCallback(async () => {
+    try {
+      await addTransactions(transactions);
+      notificationApi({
+        variant: 'success',
+        message: 'Transactions saved successfully',
+      });
+      closeModal();
+    } catch (error) {
+      notificationApi({
+        variant: 'error',
+        message: 'Error saving transactions',
+        secondaryMessage:
+          'Data extracted from the file might have duplicate item(s) with the existing data.',
+      });
+      setIsError(true);
+    }
+  }, [transactions, closeModal, addTransactions, notificationApi]);
```

Comment:
React Hook useCallback has an unnecessary dependency: 'notificationApi'. Either exclude it or remove the dependency array. Outer scope values like 'notificationApi' aren't valid dependencies because mutating them doesn't re-render the component.